File: Hosts/VS/TddStudioPackage/VsUIShellExtensions.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using Microsoft.VisualStudio.Shell.Interop;

namespace R4nd0mApps.TddStud10.Hosts.VS
{
    // TODO: Move to fs.
    public static class VsUIShellExtensions
    {
        public static MessageBoxResult DisplayMessageBox(this IVsUIShell uiShell, string title, string text)
        {
            Guid clsid = Guid.Empty;
            int result;
            Microsoft.VisualStudio.ErrorHandler.ThrowOnFailure(
              uiShell.ShowMessageBox(
                0,
                ref clsid,
                title,
                text,
                string.Empty,
                0,
                OLEMSGBUTTON.OLEMSGBUTTON_OK,
                OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST,
                OLEMSGICON.OLEMSGICON_INFO,
                0,
                out result));

            return MessageBoxResult.OK;
        }
    }
}

File: Hosts/VS/TddStudioPackage.Extensions/VsUIShellExtensions.fs
namespace R4nd0mApps.TddStud10.Hosts.VS.TddStudioPackage.Extensions

open Microsoft.VisualStudio
open Microsoft.VisualStudio.Shell.Interop
open System
open System.Runtime.CompilerServices
open System.Windows

[<Extension>]
type public VsUIShellExtensions = 
    [<Extension>]
    static member public DisplayMessageBox(uiShell : IVsUIShell, title : string, text : string) : MessageBoxResult = 
        let mutable clsid : Guid = Guid.Empty
        let mutable result : int = 0
        ErrorHandler.ThrowOnFailure
            (uiShell.ShowMessageBox
                 (0u, &clsid, title, text, String.Empty, 0u, OLEMSGBUTTON.OLEMSGBUTTON_OK, 
                  OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST, OLEMSGICON.OLEMSGICON_INFO, 0, &result)) |> ignore
        MessageBoxResult.OK
